libs/core-functions/src/lib/meta-types.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * UI for property
+ */
+export type PropertyUI = {
+  /**
+   * Optional human-friendly name of the field
+   */
+  displayName?: string;
+  /**
+   * If string field should be treated as textarea (multiline input)
+   */
+  textarea?: boolean;
+  /**
+   * If string field should be treated as password
+   */
+  password?: boolean;
+  /**
+   * If the field should not be displayed. That field must have a default value
+   */
+  hidden?: boolean | ((obj: any) => boolean);
+  /**
+   * If the field should be a constant
+   */
+  constant?: any | ((obj: any) => any);
+  /**
+   * correction to field value. e.g: set default value for property that was missing before
+   */
+  correction?: any | ((obj: any) => any);
+
+  /**
+   * Documentation for the field
+   */
+  documentation?: string;
+  /**
+   * Name of custom editor component. See getEditorComponent() function from `[workspaceId]/destinations.txt`
+   */
+  editor?:
+    | "ArrayTextarea"
+    | "StringArrayEditor"
+    | "CodeEditor"
+    | "SnippedEditor"
+    | "MultiSelectWithCustomOptions"
+    | string;
```

Comment:
I have added these types for ease of reference of the currently supported editor types but I have not removed the acceptance of `string` since there is a runtime check for the editor type and I wanted to tread lightly with this change.